scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala

```diff
@@ -50,12 +51,15 @@ object Scalafmt {
 
   private[scalafmt] def formatCode(
       code: String,
-      style: ScalafmtConfig = ScalafmtConfig.default,
+      baseStyle: ScalafmtConfig = ScalafmtConfig.default,
       range: Set[Range] = Set.empty,
-      filename: String = "<input>"
+      filename: String = defaultFilename
   ): Formatted = {
+    val style =
+      if (filename == defaultFilename) baseStyle
+      else baseStyle.getConfigFor(filename) // might throw for invalid conf
```

Comment:
put this outside of `try`, so that any problems with any config override will result in an exception on the first file.